diff --git a/dist/index.js b/dist/index.js
index 1db73afcd709b200b9762caa9b9a4cf5f5fad5ec..a3724305559de9a0e20b180cff521dd725506cb9 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -14,12 +14,12 @@ export * from "./validators.js";
 export async function createVerifiableCredentialJwt(payload, issuer, options) {
     const { network, hosts, services, timeout, verbose } = options;
     const config = mergeConfig(network, hosts, services);
-    try {
-        validateEbsiVerifiableAttestation(payload, config, timeout, verbose);
-    }
-    catch {
-        validateEbsiVerifiableAttestation202401(payload, config, timeout, verbose);
-    }
+    // try {
+    //     validateEbsiVerifiableAttestation(payload, config, timeout, verbose);
+    // }
+    // catch {
+    //     validateEbsiVerifiableAttestation202401(payload, config, timeout, verbose);
+    // }
     const { skipValidation } = options;
     if (!skipValidation) {
         const { skipAccreditationsValidation, skipStatusValidation, skipCredentialSubjectValidation, } = options;
diff --git a/dist/validators.js b/dist/validators.js
index a1ee209ff5b8666abb8f75daee03a44c23a57d15..d6f19e79130be451fa9bf1a1ffcaab90b10a96a8 100644
--- a/dist/validators.js
+++ b/dist/validators.js
@@ -72,7 +72,7 @@ function loadSchema(config, timeout) {
             return schema;
         }
         const legacyUrl = url;
-        validateTrustedSchemasRegistryUrl(legacyUrl, config);
+        // validateTrustedSchemasRegistryUrl(legacyUrl, config);
         const { data: schema } = await memoizedGet(legacyUrl, timeout);
         return schema;
     };
